### PR TITLE
Add output of the residual at the end of a time step

### PR DESCRIPTION
**Pull request by totto82**
To activte the output you will need to use
--enable-opm-rst-file=true
and
RPTRST
 CONV /

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by bska** [comment]
> ```
> RPTRST
> CONV /
> ```

The mnemonic `CONV` is **usually** reserved for something else&ndash;an integral number of times a cell has been among the "worst" `N` cells at the start of a non-linear/Newton iteration.  Would you mind considering a different mnemonic here, please?

**Comment by totto82** [comment]
Could RESIDUAL work?

**Comment by bska** [comment]
> Could `RESIDUAL` work?

Yes, that's perfect.  You'll probably have to add the mnemonic to list of [known `RPTRST` keys](https://github.com/OPM/opm-common/blob/b3177a95da0b8425b61f0acd9842549727b87909/src/opm/input/eclipse/Schedule/RSTConfig.cpp#L162) to get that through the input layer, but once that's done everything else should "just work" the way you've outlined here.

File: ebos/ecloutputblackoilmodule.hh [hunk 1]
```diff
@@ -130,6 +130,7 @@ class EclOutputBlackOilModule : public EclGenericOutputBlackoilModule<GetPropTyp
 
     enum { conti0EqIdx = Indices::conti0EqIdx };
     enum { numPhases = FluidSystem::numPhases };
+    enum { numComponents = FluidSystem::numComponents };
```
Comment (bska) [inline ebos/ecloutputblackoilmodule.hh:133]: This value is, as far as I can tell, not used anywhere in this file.  Moreover, we have the same value [in our base class](https://github.com/OPM/opm-simulators/blob/e9645ada241736af52ab5ab97a8a47a6867050e0/ebos/eclgenericoutputblackoilmodule.hh#L280).  Is there a reason you need to introduce it here?

**Review by bska — COMMENTED** [review]
This looks good for the most part.  I don't immediately see a use for the new `numComponents` value, however, so I'd like you to elaborate a little on that particular topic.  Otherwise I'm prepared to merge this.

**Comment by totto82** [comment]
jenkins build this please

**Pull request merged**